addons/http_routing/models/ir_http.py

```diff
@@ -129,37 +129,54 @@ def unslug_url(s):
 # ------------------------------------------------------------
 
 def url_for(path_or_uri, lang=None):
-    current_path = request.httprequest.path     # should already be text
+    ''' Given a relative URL, make it absolute and add the required lang or
+        remove useless lang.
+        Nothing will be done for absolute URL.
+        If there is only one language installed, the lang will not be handled
+        unless forced with `lang` parameter.
+    '''
     location = pycompat.to_text(path_or_uri).strip()
     force_lang = lang is not None
     url = werkzeug.urls.url_parse(location)
-
+    # relative URL with either a path or a force_lang
     if not url.netloc and not url.scheme and (url.path or force_lang):
-        location = werkzeug.urls.url_join(current_path, location)
+        location = werkzeug.urls.url_join(request.httprequest.path, location)
 
-        lang = pycompat.to_text(lang or request.context.get('lang') or 'en_US')
-        langs = [lg[0] for lg in request.env['ir.http']._get_language_codes()]
+        available_langs = request.env['ir.http']._get_languages()
+        if not lang:
+            lang = request.context.get('lang') or 'en_US'
+            # TODO (review): what is lang is something not available in _get_languages?
```

Comment:
use the nearest language

Reply:
Actually the previous implementation was better. I reverted that line to 
```python
lang = pycompat.to_text(lang or request.context.get('lang') or 'en_US')
```
If the lang pass to `url_for` is not a one available, the dispatcher will later correct it when accessing the link.
I don't think adding complexity here has value.

But since we now use `iso_code` (en) and not `code` (en_US) anymore for default `url_code` value, shouldn't the last condition be
```python
or 'en'
```